multi-node/aws/pkg/config/config.go

```diff
@@ -385,26 +398,32 @@ func (cfg Cluster) valid() error {
 		return fmt.Errorf("invalid vpcCIDR: %v", err)
 	}
 
-	_, instancesNet, err := net.ParseCIDR(cfg.InstanceCIDR)
-	if err != nil {
-		return fmt.Errorf("invalid instanceCIDR: %v", err)
-	}
-	if !vpcNet.Contains(instancesNet.IP) {
-		return fmt.Errorf("vpcCIDR (%s) does not contain instanceCIDR (%s)",
-			cfg.VPCCIDR,
-			cfg.InstanceCIDR,
-		)
-	}
-
 	controllerIPAddr := net.ParseIP(cfg.ControllerIP)
 	if controllerIPAddr == nil {
 		return fmt.Errorf("invalid controllerIP: %s", cfg.ControllerIP)
 	}
-	if !instancesNet.Contains(controllerIPAddr) {
-		return fmt.Errorf("instanceCIDR (%s) does not contain controllerIP (%s)",
-			cfg.InstanceCIDR,
-			cfg.ControllerIP,
-		)
+
+	for i, subnet := range cfg.Subnets {
+		if subnet.AvailabilityZone == "" {
+			return fmt.Errorf("availabilityZone must be set for subnet #%d", i)
+		}
+		_, instancesNet, err := net.ParseCIDR(subnet.InstanceCIDR)
+		if err != nil {
+			return fmt.Errorf("invalid instanceCIDR for subnet #%d: %v", i, err)
+		}
+		if !vpcNet.Contains(instancesNet.IP) {
+			return fmt.Errorf("vpcCIDR (%s) does not contain instanceCIDR (%s) for subnet #%d",
+				cfg.VPCCIDR,
+				cfg.InstanceCIDR,
+				i,
+			)
+		}
+		if i == 0 && !instancesNet.Contains(controllerIPAddr) {
```

Comment:
I have the same feeling about your concern .
Also, at this point, a separate issue sounds O.K. for me to keep us proceeding incrementally.